packages/web-api/src/methods.ts

```diff
@@ -2104,12 +2104,14 @@ export interface UsersProfileSetArguments extends WebAPICallOptions, TokenOverri
 }
 
 export interface ViewsOpenArguments extends WebAPICallOptions, TokenOverridable {
-  trigger_id: string;
```

Comment:
@srajiang - In order to accept either `trigger_id` or `interactivity_pointer` I made them both optional - is there a better way to handle this or should this work? Ideally it would require at least one of these

Reply:
@hello-ashleyintech  I think it would still be good to require at least one. So setting both to optional might let devs fail by setting neither. 

What do you think about logical XOR, so that it must be either trigger_id OR interactivity but not both or neither? There seem to be a [few ways folks in SO have discussed tackling this](https://stackoverflow.com/questions/44425344/typescript-interface-with-xor-barstring-xor-cannumber) which might be a good resource for this.

Reply:
@srajiang took a look at this approach and it seems like the XOR will work but only by declaring a new type alias to populate the union into, meaning that the output type of `ViewsOpenArguments` and `ViewsPushArguments` would be `type` instead of `interface`. will this pose any issues? it looks like the primary differences with types and interfaces are that types can't be changed after being created.

Reply:
@hello-ashleyintech Yeah, so I guess if we took this route, the dev wouldn't be able to say, import that type and extend it using the `extends` TS syntax. Or internally, we couldn't use the `extends` syntax to extend a new interface out from it. 

But they they could declare a new type from there and use that imported type in an intersection or something if they wanted to add more attributes..

I haven't yet come across a scenario that required only `interface` vs. `type` so afaik that change could be fine.